system_tests/snmp_proxy.py

```diff
@@ -0,0 +1,59 @@
+from os import path
+
+from influxdb import InfluxDBClient
+
+from cosmo_tester.framework.testenv import TestCase
+
+
+class TestSNMPProxy(TestCase):
+    TIME_TO_WAIT_FOR_METRICS = 10  # in seconds
+
+    def _check_influx_db(self, deployment_id):
+        client = InfluxDBClient(self.env.management_ip, database='cloudify')
+        try:
+            res = client.query(
+                'select * from /^{0}\./i where time > now() - {1}s'
+                .format(deployment_id, self.TIME_TO_WAIT_FOR_METRICS)
+            )
+        except NameError as e:
+            self.fail('Monitoring events for deployment with ID {0} have'
+                      ' not been found in influxDB. The error is: {1}'
+                      .format(deployment_id, e))
+
+        self.assertTrue(res[0]['points'])  # Assert not empty
+
+    def _test_snmp_monitoring(self, blueprint, inputs):
+        self.blueprint_yaml = path.join(
+            path.dirname(path.dirname(path.realpath(__file__))),
+            blueprint
+        )
+        deployment_id = self.test_id
+        self.upload_deploy_and_execute_install(
+            inputs=inputs,
+            deployment_id=deployment_id
+        )
+
+        self.repetitive(
+            self._check_influx_db,
+            self.TIME_TO_WAIT_FOR_METRICS,
```

Comment:
Do you think 10 seconds is enough? If there is change for sporadic failure because it may not be enough, then it should be increased. It shouldn't affect the mean run time of this test as in the common case it will take the minimum required time and only when it fails for some reason will it take the set timeout. 
Also, consider using a keyword argument for timeout.